Use variables for the numbers in the markdown text

turnerm opened the pull request.

joseepoirier commented on automated_reports/report_generation_AFG.Rmd:
```diff
@@ -48,26 +48,39 @@ editor_options:
-* The trajectory of the crisis depends on whether NPIs are kept in place: lifting of NPIs would lead daily new cases to hold steady or increase rather than decrease if NPIs are maintained. Thus lifting NPIs would result in an increase in cases and deaths by up to 598 more cases and 42 more deaths (see sections 2 and 3 for details). Due to the lag between cases and deaths, a larger number of cases ...
-* The number of active severe cases requiring healthcare is projected to decrease but still reach 557 - 752 if NPIs are maintained. Hospitalizations may increase if NPIs are lifted (see section 3).
+
+* In 4 weeks, we project {npi_projected_deltas_cases_4w_LOW} - {npi_projected_deltas_cases_4w_HI} additional cases and {npi_projected_deltas_deaths_4w_LOW} - {npi_projected_deltas_deaths_4w_LOW} additional deaths will be reported if current NPIs are maintained.
```
The second {npi_projected_deltas_deaths_4w_LOW}  should be {npi_projected_deltas_deaths_4w_HI}

turnerm replied:
Good catch, fixed!

joseepoirier commented on automated_reports/report_generation_AFG.Rmd:
```diff
@@ -48,26 +48,39 @@ editor_options:
+
+* In 4 weeks, we project {npi_projected_deltas_cases_4w_LOW} - {npi_projected_deltas_cases_4w_HI} additional cases and {npi_projected_deltas_deaths_4w_LOW} - {npi_projected_deltas_deaths_4w_LOW} additional deaths will be reported if current NPIs are maintained.
+
+* The trajectory of the crisis depends on whether NPIs are kept in place: lifting of NPIs would lead daily new cases to hold steady or increase rather than decrease if NPIs are maintained. Thus lifting NPIs would result in an increase in cases and deaths by up to {no_npi_projected_deltas_cases_4w_MAX} more cases and {no_npi_projected_deltas_deaths_4w_MAX} more deaths (see sections 2 and 3 for det ...
```
I think these are the metrics you are looking for: Maximum number of   extra cases if NPIs are lifted and 
Maximum number of extra deaths if NPIs   are lifted

turnerm replied:
I think that's what is already here, the variable is defined as: 
```R
no_npi_projected_deltas_cases_4w_MAX <- report_metrics %>% filter(metric_name == "Maximum number of extra cases if NPIs are lifted") %>% select(metric_value) %>% as.numeric()
```

joseepoirier commented on automated_reports/report_generation_AFG.Rmd:
```diff
@@ -48,26 +48,39 @@ editor_options:
+
+* The trajectory of the crisis depends on whether NPIs are kept in place: lifting of NPIs would lead daily new cases to hold steady or increase rather than decrease if NPIs are maintained. Thus lifting NPIs would result in an increase in cases and deaths by up to {no_npi_projected_deltas_cases_4w_MAX} more cases and {no_npi_projected_deltas_deaths_4w_MAX} more deaths (see sections 2 and 3 for det ...
+
+* The number of active severe cases requiring healthcare is projected to decrease but still reach {npi_hospitalizations_LOW} - {npi_hospitalizations_HI} if NPIs are maintained. Hospitalizations may increase if NPIs are lifted (see section 3).
```
Parts of the key messages change from report-to-report (between countries and over time). Be very careful not to copy/paste key messages from one country to another, and to check that the metrics you included are on a given week the clearest way to communicate the situation. This is an example of that need to remain flexible: sometimes I'd report the total projected hospitalizations, sometimes the projected delta in hospitalizations. The word "decrease" here (and "increase" in the next sentence) should also be conditional on the sign of the delta. I used the total in this case because there was a slight decrease projected but the hospitalizations remained high. If I had left only the "decreasing hospitalization"/delta phrasing, it could have been somewhat misleading the reader into thinking hospitalizations were no longer a strain on healthcare resources.

turnerm replied:
Yes, very good point!

joseepoirier commented on automated_reports/report_generation_AFG.Rmd:
```diff
@@ -179,6 +193,12 @@ no_npi_projected_deltas_deaths_4w_LOW <- format(round(no_npi_projected_deltas_de
 no_npi_projected_deltas_deaths_4w_HI <- report_metrics %>% filter(metric_name == "NO NPI - projected additional reported deaths in 4w - MAX") %>% select(metric_value) %>% as.numeric()
 no_npi_projected_deltas_deaths_4w_HI <- format(round(no_npi_projected_deltas_deaths_4w_HI, 0), big.mark = ",")
 
+no_npi_projected_deltas_cases_4w_MAX <- report_metrics %>% filter(metric_name == "Maximum number of extra cases if NPIs are lifted") %>% select(metric_value) %>% as.numeric()
```
I'd recommend adding a rounding steps after numeric. I see you selected the correct metric here; it would be cleaner to change the name of the variables to better describe what it is (it's not a delta. It's the difference between the highest value of the HI range minus the lowest value of the LOW range.)

turnerm replied:
Good point. I'd argue that it's still a delta in the sense that it's dealing with additional cases, which is how "delta" is used in the other variable names. What do you think about:
`no_npi_vs_npi_projected_deltas_cases_4W_MAX`

joseepoirier commented on automated_reports/report_generation_AFG.Rmd:
```diff
@@ -179,6 +193,12 @@ no_npi_projected_deltas_deaths_4w_LOW <- format(round(no_npi_projected_deltas_de
+no_npi_projected_deltas_cases_4w_MAX <- report_metrics %>% filter(metric_name == "Maximum number of extra cases if NPIs are lifted") %>% select(metric_value) %>% as.numeric()
+no_npi_projected_deltas_cases_4w_MAX <- format(round(no_npi_projected_deltas_cases_4w_MAX, 0), big.mark = ",")
+
+no_npi_projected_deltas_deaths_4w_MAX <- report_metrics %>% filter(metric_name == "Maximum number of extra deaths if NPIs are lifted") %>% select(metric_value) %>% as.numeric()
```
Same as above. The MAX suffix does help, though.

joseepoirier commented on automated_reports/report_generation_AFG.Rmd:
```diff
@@ -192,12 +212,24 @@ npi_hospitalizations_LOW <- format(round(npi_hospitalizations_LOW, 0), big.mark
 npi_hospitalizations_HI <- format(round(npi_hospitalizations_HI, 0), big.mark = ",")
 
+npi_hospitalizations_delta_LOW <- report_metrics %>% filter(metric_name == "NPI additional Hospitalizations projections 4w - MIN") %>% select(metric_value) %>% as.numeric()
+npi_hospitalizations_delta_LOW <- format(round(abs(npi_hospitalizations_delta_LOW), 0), big.mark = ",")
```
Is ABS introducing the risk of misrepresenting the trend? Like the "decrease" word above not being selected conditionally on the sign of the variable?

turnerm replied:
Good point. The problem is that you don't want the negative sign in the text, but I think that should be dealt with in the markdown section itself so that it's obvious if the sign flips.

joseepoirier reviewed (commented):
Neat way to limit manual data entry! The key messages were evolving too often back in the Fall but I had been fairly stable in the last 1-2 months so this approach makes sense. 

I would still recommend taking a close look at the numbers to ensure the right metric is selected to convey the situation as we see it overall (deltas vs. totals for example, avoid saying a projected range of 4-4 extra deaths, etc.) 
Nicely done!

turnerm commented:
> Parts of the key messages change from report-to-report (between countries and over time). Be very careful not to copy/paste key messages from one country to another, and to check that the metrics you included are on a given week the clearest way to communicate the situation. This is an example of that need to remain flexible: sometimes I'd report the total projected hospitalizations, sometimes the projected delta in hospitalizations. The word "decrease" here (and "increase" in the next sentence) should also be conditional on the sign of the delta. I used the total in this case because there was a slight decrease projected but the hospitalizations remained high. If I had left only the "decreasing hospitalization"/delta phrasing, it could have been somewhat misleading the reader into thinking hospitalizations were no longer a strain on healthcare resources.

Quoting this again because I think it's very important! I hope that with these changes, I can use the time saved by not need to copy-paste to instead focus on getting the key messages updated based on any changes to the situation (so always making sure to look at the new numbers and plots carefully, and comparing to the previous report), and making sure that the numbers are correct.

turnerm commented:
@joseepoirier thanks so much for the review, I've addressed your comments, let me know if you agree with the changes and have anything further to add!

joseepoirier commented on automated_reports/report_generation_AFG.Rmd:
```diff
@@ -48,26 +48,39 @@ editor_options:
-* The trajectory of the crisis depends on whether NPIs are kept in place: lifting of NPIs would lead daily new cases to hold steady or increase rather than decrease if NPIs are maintained. Thus lifting NPIs would result in an increase in cases and deaths by up to 598 more cases and 42 more deaths (see sections 2 and 3 for details). Due to the lag between cases and deaths, a larger number of cases ...
-* The number of active severe cases requiring healthcare is projected to decrease but still reach 557 - 752 if NPIs are maintained. Hospitalizations may increase if NPIs are lifted (see section 3).
+
+* In 4 weeks, we project {npi_projected_deltas_cases_4w_LOW} - {npi_projected_deltas_cases_4w_HI} additional cases and {npi_projected_deltas_deaths_4w_LOW} - {npi_projected_deltas_deaths_4w_HI} additional deaths will be reported if current NPIs are maintained.
```
would "we project that" read more eaily?

turnerm replied:
Yes good point, fixed!

joseepoirier commented on automated_reports/report_generation_AFG.Rmd:
```diff
@@ -179,6 +193,12 @@ no_npi_projected_deltas_deaths_4w_LOW <- format(round(no_npi_projected_deltas_de
 no_npi_projected_deltas_deaths_4w_HI <- report_metrics %>% filter(metric_name == "NO NPI - projected additional reported deaths in 4w - MAX") %>% select(metric_value) %>% as.numeric()
 no_npi_projected_deltas_deaths_4w_HI <- format(round(no_npi_projected_deltas_deaths_4w_HI, 0), big.mark = ",")
 
+no_npi_vs_npi_projected_deltas_cases_4W_MAX <- report_metrics %>% filter(metric_name == "Maximum number of extra cases if NPIs are lifted") %>% select(metric_value) %>% as.numeric()
```
Ln1960197 should be MIN, right?

turnerm replied:
For this metric, we only use the maximum value in the text

joseepoirier replied:
Ah yes! Sorry I saw two MAX in a row and it broke the MIN-MAX pattern. Yes those two should be MAX.

joseepoirier reviewed (approved):
Seems to be 1 error and I had one minor suggestion. Didn't review each Rmd let me know if you'd like me to.
